packages/core/lib/package.js

```diff
@@ -210,169 +209,136 @@ const Package = {
+            } catch (error) {
+              reject(error);
             }
-
-            resolve(body);
-          }
+          })
         );
-      }));
-
-      var contract_types = {};
-      var deployments = {};
-
-      Promise.all(promises)
-        .then(function(contracts) {
-          // contract_types first.
-          contracts.forEach(function(data) {
-            contract_types[data.contractName] = {
-              contract_name: data.contractName,
-              bytecode: data.bytecode,
-              abi: data.abi
-            };
-          });
-
-          //var network_cache = {};
-          var matching_promises = [];
-
-          contracts.forEach(function(data) {
-            Object.keys(data.networks).forEach(function(network_id) {
-              matching_promises.push(
-                new Promise(function(accept, reject) {
-                  // Go through each deployed network and see if this network matches.
-                  // Bail early if we foun done.
-                  async.each(
-                    deployed_networks,
-                    function(deployed_network, finished) {
-                      Networks.matchesNetwork(
-                        network_id,
-                        options.networks[deployed_network],
-                        function(err, matches) {
-                          if (err) return finished(err);
-                          if (matches) {
-                            var uri = uris[deployed_network];
-
-                            if (!deployments[uri]) {
-                              deployments[uri] = {};
-                            }
-
-                            deployments[uri][data.contractName] = {
-                              contract_type: data.contractName, // TODO: Handle conflict resolution
-                              address: data.networks[network_id].address
-                            };
-
-                            return finished("bail early");
-                          }
-                          finished();
-                        }
-                      );
-                    },
-                    function(err) {
-                      if (err && err !== "bail early") {
-                        return reject(err);
-                      }
-
-                      accept();
-                    }
-                  );
-                })
-              );
-            });
-          });
-
-          return Promise.all(matching_promises);
-        })
-        .then(function() {
-          var to_return = {
-            contract_types: contract_types,
-            deployments: deployments
-          };
-          if (callbackPassed) {
-            callback(null, to_return);
-            return;
-          }
-          return to_return;
-        })
-        .catch(error => {
-          if (callbackPassed) {
-            callback(error);
-          }
-          throw error;
-        });
+      });
     });
+
+    Promise.all(matchingPromises)
```

Comment:
Since publishable artifacts is `async` now we can just await this `Promise.all` and simplify this whole block.

Reply:
Yeah, sounds good to me!!